src/core/unit.c

```diff
@@ -5842,6 +5842,9 @@ static int unit_freezer_action(Unit *u, FreezerAction action) {
         if (s != UNIT_ACTIVE)
                 return -EHOSTDOWN;
 
+        if (u->pending_freezer_message)
+                return -EALREADY;
+
```

Comment:
I am just wondering why check that is immediately below is not enough.

Reply:
Indeed. When pending_freezer_message is set, then the state must be FREEZING or THAWING.